tutorial/tutorial.md

```diff
@@ -0,0 +1,333 @@
+  "modified_at": "2023-03-03T15:44:56.505522+01:00",
+  "name": "lmolteni-6225FC431494432",
+  "annotations": {
+    "cos.bf2.org/organisation-id": "432432"
+  },
+  "status": {
+    "state": "disconnected"
+  }
+}
+```
+
+
+The creation of the cluster also creates a service account with the name of the cluster, if you want to check the created service-account you can use `rhoas`
+
+`rhoas service-account list`
+
+The cluster is in `disconnected` state. This means the control plane and the cluster aren't actively communicating.
+We'll fix it soon, but first let's generate a few secrets and insert them in the local cluster. 
+
+Gather the id from the previous request (in this case `cg10fpruns2rbjn4ka20`) and run
+
+```
+kubectl config set-context --current --namespace=redhat-openshift-connectors
+
+bin/create-cluster-secret <CLUSTER_ID>`
+```
+
+The first line is needed as `create-cluster-secret` needs to know the namespace where to create the secret.
+In this secret there will be a few parameters such as the `clientId` and the `clientSecret` of the service account user
+created during the cluster creation. Such parameters will be used by the `cos-fleetshard-sync` to communicate with the control plane. 
+
+To verify the secret was created accordingly, run
+
+`kubectl -n redhat-openshift-connectors get secret -o yaml`
+
+In another tab, run the `cos-fleet-shard-sync` module.
+
+```
+cd cos-fleetshard/cos-fleetshard-sync
+mvn -Dlocal -Dcontrol-plane-base-url=http://localhost:8000
+```
+
+The `control-plane-base-url` should be taken from the secrets created previously, but due to a bug it's being set to `localhost` instead. 
+Force it to `http://localhost:8000` to use the local control plane running on the environment.
+
+On the `cos-tools` tab, now check the the cluster is in `ready` state. There should be no errors in the `cos-fleetshard-sync` mvn run.
+
+```
+❯ ./get-clusters
+ID             NAME                    OWNER                STATUS
+--             ----                    -----                ------
+cg10fpruns2rb  lmolteni-6225FC4314944  lmolteni             ready
+```
+
+# Create a local catalogue
+
+To use camel connectors, RHOC control plane needs to know the existing available connectors.  
+
+Make a folder in `kas-fleet-manager` called `catalog` (it will be ignored by `.gitgnore` eventually, otherwise add the folder to `.gitignore`)
+
+Get the connectors metadata
+
+Create a new file in `cos-fleet-manager/catalog/cos-fleet-catalog-metadata/metadata.yaml`
+Copy [this](https://github.com/manstis/cos-fleet-manager/blob/main/templates/connector-metadata-camel-template.yaml#L21-L485) into the file (format it accordingly, as we love YAML syntax errors)
+
+Get the connector definitions
+
+`git clone https://github.com/bf2fc6cc711aee1a0c2a/cos-manifests`
+
+Copy `cos-manifests/connectors/cos-fleet-catalog-camel` to the `cos-fleet-manager/catalog` folder
+
+Delete the `cos-fleet-manager/catalog/cos-fleet-catalog-camel/connectors.json` file
```

Comment:
maybe you can just point to the location of the cloned repo and point to the individual catalogs instead of copy and delete so you can update the catalog with git pull

Reply:
We only need a [part](https://github.com/manstis/cos-fleet-manager/blob/main/templates/connector-metadata-camel-template.yaml#L21-L485) of the `connector-camel-template.yaml` file: